test/v2/controllers/validators/DeleteHistoricUkPropertyAnnualSubmissionValidatorFactorySpec.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers.validators
+
+import api.models.domain.{HistoricPropertyType, Nino, TaxYear}
+import api.models.errors._
+import mocks.MockAppConfig
+import support.UnitSpec
+import v2.models.request.deleteHistoricUkPropertyAnnualSubmission.DeleteHistoricUkPropertyAnnualSubmissionRequestData
+
+class DeleteHistoricUkPropertyAnnualSubmissionValidatorFactorySpec extends UnitSpec with MockAppConfig {
+
+  implicit val correlationId: String = "X-123"
+  private val validNino              = "AA123456A"
+  private val validTaxYear           = "2021-22"
+  private val propertyType           = HistoricPropertyType.Fhl
+
+  private val parsedNino    = Nino(validNino)
+  private val parsedTaxYear = TaxYear.fromMtd(validTaxYear)
+
+  private val validatorFactory = new DeleteHistoricUkPropertyAnnualSubmissionValidatorFactory(mockAppConfig)
+
+  private def validator(nino: String, taxYear: String, propertyType: HistoricPropertyType) = validatorFactory.validator(nino, taxYear, propertyType)
+
+  MockAppConfig.minimumTaxHistoric.returns(2020)
+  MockAppConfig.maximumTaxHistoric.returns(2023)
+
+  "validator" should {
+    "return no errors" when {
+      "a valid request is supplied" in {
```

Comment:
Might be worth splitting this into two valid test cases - one for FHL and one for non-fhl where the propertyType is passed in explicitly